### ISSUE TITLE
Settings/Config not persist a reboot

### ISSUE THREAD

**Issue by vk496** [chenxiaolong/BCR#598]
Hello,

I noticed that BCR settings are not surviving a phone reboot (including the "enable record" switch). They just get lost, like if I installed it for the first time.

I have no idea how can I provide more debug logs of CBR

Android 14
Magisk 27004 +Zygdisk +Ramdisk

**Comment by chenxiaolong** [comment]
I suspect this is related to BCR's recent feature where it can record calls from before the device is initially unlocked after a reboot.

2 other folks have reported this (#582 and #591). Android has 2 data directories for apps, `/data/user_de/0/<app>` (available before unlock) and `/data/data/<app>` (available after unlock). BCR stores its settings in the first folder. It seems that on some devices, that is either getting wiped by Android or Android is not allowing it to be written to in the first place. I have no idea why that's happening or how to work around it.

---

Could you please post the output of this?
```bash
adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
```

And then after a reboot, but before unlocking, please grab a logcat and upload it too:
```bash
adb logcat -d > logcat.txt
```

**Comment by vk496** [comment]
```
$ adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
drwx------ 6 u0_a335 u0_a335 u:object_r:privapp_data_file:s0:c512,c768       3440 2024-07-13 02:21 /data/data/com.chiller3.bcr
drwx------ 4 u0_a335 u0_a335 u:object_r:app_data_file:s0:c79,c257,c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr
```

Reboot and not unlocking the phone:

[logcat.txt](https://github.com/user-attachments/files/16923301/logcat.txt)

**Comment by vk496** [comment]
And status after the reboot (before unlock):

```
$ adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
ls: /data/data/com.chiller3.bcr: No such file or directory
drwx------ 4 u0_a335 u0_a335 u:object_r:app_data_file:s0:c79,c257,c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr
```

**Comment by vk496** [comment]
Phone: OnePlus 14

**Comment by chenxiaolong** [comment]
Hmm, the `u:object_r:app_data_file:s0:c79,c257,c512,c768` SELinux label for `/data/user_de/0/com.chiller3.bcr` is definitely not right. It should be `u:object_r:privapp_data_file:s0:c512,c768`, just like the other folder. I wonder if OxygenOS is doing something weird (all reports are with OnePlus devices).

Could you give this a try?
```bash
adb shell su -c 'restorecon -RDv /data/user_de/0/com.chiller3.bcr'
```

If it changed anything, there should be output like:
```
SELinux:  Relabeling /data/user_de/0/com.chiller3.bcr from u:object_r:app_data_file:s0:c79,c257,c512,c768 to u:object_r:privapp_data_file:s0:c512,c768.
```

If that works, I can it to BCR's boot script.

**Comment by vk496** [comment]
```
PS C:\Users\vk496> adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
* daemon not running; starting now at tcp:5037
* daemon started successfully
drwx------ 6 u0_a335 u0_a335 u:object_r:privapp_data_file:s0:c512,c768       3440 2024-07-13 02:21 /data/data/com.chiller3.bcr
drwx------ 4 u0_a335 u0_a335 u:object_r:app_data_file:s0:c79,c257,c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr

# reboot

PS C:\Users\vk496> adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
drwx------ 4 u0_a335 u0_a335 u:object_r:app_data_file:s0:c79,c257,c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr
ls: /data/data/com.chiller3.bcr: No such file or directory

PS C:\Users\vk496> adb shell su -c 'restorecon -RDv /data/user_de/0/com.chiller3.bcr'
SELinux: Loaded file context from:
                /system/etc/selinux/plat_file_contexts
                /system_ext/etc/selinux/system_ext_file_contexts
                /product/etc/selinux/product_file_contexts
                /vendor/etc/selinux/vendor_file_contexts
SELinux:  Relabeling /data/user_de/0/com.chiller3.bcr from u:object_r:app_data_file:s0:c79,c257,c512,c768 to u:object_r:privapp_data_file:s0:c512,c768.
SELinux:  Relabeling /data/user_de/0/com.chiller3.bcr/cache from u:object_r:app_data_file:s0:c79,c257,c512,c768 to u:object_r:privapp_data_file:s0:c512,c768.
SELinux:  Relabeling /data/user_de/0/com.chiller3.bcr/code_cache from u:object_r:app_data_file:s0:c79,c257,c512,c768 to u:object_r:privapp_data_file:s0:c512,c768.

PS C:\Users\vk496> adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
ls: /data/data/com.chiller3.bcr: No such file or directory
drwx------ 4 u0_a335 u0_a335 u:object_r:privapp_data_file:s0:c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr

# Unlock

PS C:\Users\vk496> adb shell su -c 'ls -ldZ /data/user_de/0/com.chiller3.bcr /data/data/com.chiller3.bcr'
drwx------ 6 u0_a335 u0_a335 u:object_r:privapp_data_file:s0:c512,c768  3440 2024-07-13 02:21 /data/data/com.chiller3.bcr
drwx------ 4 u0_a335 u0_a335 u:object_r:privapp_data_file:s0:c512,c768  3440 2024-07-13 02:19 /data/user_de/0/com.chiller3.bcr
```


HOWEVER, right now I have the config surviving a reboot. Even when I don't run that command

**Comment by chenxiaolong** [comment]
Thanks for testing that! Good to know that it persists.

I've added this workaround in #600, which will be included in BCR 1.70.
